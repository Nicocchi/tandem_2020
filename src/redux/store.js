import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import combineReducer from "./reducers";

let store = null;

// Redux dev tools
const composeEnhancers =
    typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
        ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
              // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
          })
        : compose;

const devEnhancer = composeEnhancers(applyMiddleware(thunk, logger));
const enhancer = composeEnhancers(applyMiddleware());

// Create redux store
if (process.env.REACT_APP_NODE_ENV !== "production") {
    store = createStore(combineReducer, devEnhancer);
} else {
    store = createStore(combineReducer, enhancer);
}

export default store;