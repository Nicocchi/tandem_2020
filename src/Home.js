import React, { useEffect } from "react";
//import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import { startGame, endGame } from "./redux/actions";
import { connect } from "react-redux";

function Home(props) {
    useEffect(() => {
        props.endGame();
        return () => {};
    });
    return (
        <header className="App-header">
            <img alt="Tandem Trivia" src="images/logo.png" />
            <NavLink to="/play" onClick={props.startGame} className="play-btn">
                Play
            </NavLink>
        </header>
    );
}

//Home.propTypes = {};

export default connect(null, { startGame, endGame })(Home);
