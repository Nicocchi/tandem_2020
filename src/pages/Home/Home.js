import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import { connect } from "react-redux";
import { startGame, endGame } from "./../../redux/actions";

/**
 * Home page
 *
 * @param {*} props
 * @returns
 */
function Home(props) {
    useEffect(() => {
        props.endGame();
        return () => {};
    });
    return (
        <header className="App-header">
            <p className="text-shadow-pop-left title1">Tandem</p>
            <p className="text-shadow-pop-left title2">Trivia</p>
            <NavLink to="/play" onClick={props.startGame} className="play-btn animated-infinite swing">
                Play
            </NavLink>
        </header>
    );
}

Home.propTypes = {
    startGame: PropTypes.func
};

export default connect(null, { startGame, endGame })(Home);
