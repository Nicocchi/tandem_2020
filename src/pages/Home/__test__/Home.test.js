import React from "react";
import ReactDOM from "react-dom";
import Home from "./../Home";
import store from "../../../redux/store";
import { Provider } from "react-redux";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

import { cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

describe("Home Page", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });

    afterEach(() => {
        cleanup();
    });

    it("renders without crashing without props", () => {
        const div = document.createElement("div");

        ReactDOM.render(
            <Provider store={store}>
                <Router history={history}>
                    <Home history={history} />
                </Router>
            </Provider>,
            div
        );
    });

    it("matches snapshop", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <Router history={history}>
                        <Home history={history} />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
