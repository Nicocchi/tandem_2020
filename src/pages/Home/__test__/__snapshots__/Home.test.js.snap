// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches snapshop 1`] = `
<header
  className="App-header"
>
  <p
    className="text-shadow-pop-left title1"
  >
    Tandem
  </p>
  <p
    className="text-shadow-pop-left title2"
  >
    Trivia
  </p>
  <a
    aria-current={null}
    className="play-btn animated-infinite swing"
    href="/play"
    onClick={[Function]}
  >
    Play
  </a>
</header>
`;
